docs/api_docs/maintenance/scale/horizontal-scale.md

```diff
@@ -0,0 +1,467 @@
+---
+title: Horizontal Scale
+description: How to scale a cluster, scale replicas, scale instances
+keywords: [horizontal scaling, Horizontal Scale]
+sidebar_position: 2
+sidebar_label: Horizontal Scale
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Horizontal Scale
+
+From v0.9.0, the horizontal scale can be divided into two types to better support various scenarios. Both final-state-oriented and procedure-oriented scales support scale in and out, but what distinguishes them is that the final-state-oriented scale only scales replicas and the procedure-oriented one supports scaling both the replicas and instances.
+
+- Final-state-oriented
+
+    Final-state-oriented horizontal scale refers to the operation in which components and instances specify replicas. This type of horizontal scale is performed by simply specifying the replica amount and overwriting the replicas. However, this method does not support specifying an instance to scale in or out.
+
+    It is also recommended to edit the cluster YAML file to perform a horizontal scale since its corresponding OpsRequest option might be deprecated later.
+
+- Procedure-oriented
+
+    The procedure-oriented horizontal scale is designed to add or delete specified instances to meet the frequent scaling in and out demands.
+
+    For the procedure-oriented horizontal scale, an operation will calculate the horizontal scale state based on the status of the pods involved in the OpsRequest operation. In extreme cases, due to the non-atomic nature of the operation, the recorded lastCompReplicas may be inaccurate, leading to incorrect pod count calculations (you can ensure the order on the client side or set `KUBEBLOCKS_RECONCILE_WORKERS=1`).
+
+     1. If the issued OpsRequest attempts to delete an instance created by a running OpsRequest, it will not be allowed and will fail directly.
+     2. If there is a final-state-oriented operation in execution, this operation will be terminated.
```

Comment:
If there is an ongoing scaling operation using the declarative API, this operation will be terminated.